core/model/src/main/java/org/eclipse/rdf4j/model/util/RDFContainers.java

```diff
@@ -0,0 +1,397 @@
+			Consumer<Statement> consumer,
+			Resource... contexts) {
+		consumeContainer(containerType, values, container, consumer, SimpleValueFactory.getInstance(), contexts);
+	}
+
+	/**
+	 * Converts the supplied {@link Iterable} to an <a href="https://www.w3.org/TR/rdf-schema/#ch_container">RDF
+	 * Container</a>, using the supplied {@code head} resource as the starting resource of the RDF Container. The
+	 * statements making up the new RDF Container will be reported to the supplied {@link Consumer} function.
+	 *
+	 * @param containerType defines the type of RDF Container
+	 * @param values        an {@link Iterable} of objects (such as a Java {@link Collection} ), which will be converted
+	 *                      to an RDF Container. May not be {@code null}. The method attempts to convert each value that
+	 *                      is not already an instance of {@link Value} to a {@link Literal}. This conversion will fail
+	 *                      with a {@link LiteralUtilException} if the value's object type is not supported. See
+	 *                      {@link Literals#createLiteralOrFail(ValueFactory, Object)} for an overview of supported
+	 *                      types.
+	 * @param container     a {@link Resource} which will be used as the head of the container, that is, the starting
+	 *                      point of the created RDF Container. May be {@code null}, in which case a new resource is
+	 *                      generated to represent the containter head.
+	 * @param consumer      the {@link Consumer} function for the Statements of the RDF Container. May not be
+	 *                      {@code null}.
+	 * @param vf            the {@link ValueFactory} to use for creation of new model objects. May not be {@code null}
+	 * @param contexts      the context(s) in which to add the RDF Container. This argument is an optional vararg and
+	 *                      can be left out.
+	 * @throws LiteralUtilException if one of the supplied values can not be converted to a Literal.
+	 * @see <a href="https://www.w3.org/TR/rdf-schema/#ch_container">RDF Schema 1.1 section on Collection
+	 *      vocabulary</a>.
+	 * @see Literals#createLiteralOrFail(ValueFactory, Object)
+	 *
+	 * @since 3.0
+	 */
+	public static void consumeContainer(IRI containerType, Iterable<?> values, Resource container,
+			Consumer<Statement> consumer,
+			ValueFactory vf, Resource... contexts) {
+		Objects.requireNonNull(values, "input collection may not be null");
+		Objects.requireNonNull(consumer, "consumer may not be null");
+		Objects.requireNonNull(vf, "injected value factory may not be null");
+
+		Resource current = container != null ? container : vf.createBNode();
+		boolean validType = Objects.equals(containerType, RDF.ALT) ||
+				Objects.equals(containerType, RDF.BAG) ||
+				Objects.equals(containerType, RDF.SEQ);
+
+		if (!validType) {
+			throw new RuntimeException("containerType should be one of ALT, BAG or SEQ");
```

Comment:
I think this should be a `ModelException`, rather than a generic `RuntimeException`.